packages/core/nuxt-theme-module/theme/pages/Category.vue

```diff
@@ -440,6 +443,26 @@ export default {
       changeFilters(selectedFilters.value);
     };
 
+    const handleAddItemToCart = async (product, quantity) => {
+      await addItemToCart({ product, quantity });
```

Comment:
Wondering if it wouldn't be easier to use `then` `catch` here? wdyt @andrzejewsky @Fifciu @lukaszjedrasik ?

Reply:
Indeed we could use `then`. But we could not use `catch` as the method invocation is already inside try...catch block without error hoisting:
```js
const addItem = async ({ product, quantity, customQuery }) => {
      Logger.debug('useCart.addItem', { product, quantity });

      try {
        loading.value = true;
        error.value.addItem = null;
        const updatedCart = await factoryParams.addItem(
          context,
          {
            currentCart: cart.value,
            product,
            quantity,
            customQuery
          }
        );
        cart.value = updatedCart;
      } catch (err) {
        error.value.addItem = err;
        Logger.error('useCart/addItem', err);
      } finally {
        loading.value = false;
      }
    };
```

So `catch` would be never called.

Reply:
perhaps we should throw errors as well then here